Assets/_Functionalities/Wms/Scripts/WmsGetCapabilities.cs

```diff
@@ -184,13 +185,32 @@ public List<MapFilters> GetMaps(int width, int height, bool transparent)
                 // Extract styles for the layer
                 var styles = ExtractStyles(mapNode);
 
-                // CRS/SRS may be defined in the current MapNode, but can also inherit from a parent if it is not
-                // specified the flag at the end of this function will check the current node and its parents
-                var spatialReference = GetInnerTextForNode(mapNode, mapTemplate.spatialReferenceType, true);
-                
-                // TODO: Really ugly fix to deal issues around EPSG:4326. So we fixate on CRS:84 now, hoping that will
-                // work in all situations
-                spatialReference = "CRS:84";
+                string spatialReference = null;
+                var spatialReferenceType = MapFilters.SpatialReferenceTypeFromVersion(new Version(mapTemplate.version));
+                XmlNodeList crsNodes = GetNodesByName(mapNode, spatialReferenceType);
+                if (crsNodes.Count > 0)
+                { 
+                    for (int i = 0; i < preferredCRS.Length; i++)
+                    {
+                        foreach (XmlNode crsNode in crsNodes)
+                        {
+                            if (preferredCRS[i] == crsNode.InnerText)
+                            {
+                                spatialReference = crsNode.InnerText;
+                                goto referenceFound;
+                            }
+                        }
+                    }
+                    if (string.IsNullOrEmpty(spatialReference))
+                    {
+                        spatialReference = crsNodes[0].InnerText;
```

Comment:
Dit geeft niet de eerste _ondersteunde_ CRS terug maar de eerste in de set crsNodes, zie AC 3. Dit kan een ongeldige situatie opleveren